### DISCUSSION TITLE
Generated Apis dont use the Refit Generator

### DISCUSSION THREAD

**Discussion by Codelisk** [reactiveui/refit#1600] [Q&A] [answered]
Hello,

when I generate my Own Apis, the Refit Code Generator is not triggered. Is there a way to workaround that?

If you want to elaborate I could also try to create a sample for it.

Thank you very much in advance.

**Comment by glennawatson** [comment, marked as answer]
The Refit source generator only runs against interfaces that:
- live in a project that references the `Refit` package (so the generator is loaded), and
- have at least one method decorated with a Refit HTTP method attribute (`[Get]`, `[Post]`, etc.).

Common reasons "my own APIs don't trigger the generator":
- The interface is defined in a different assembly/project that does not reference Refit, so the generator never sees it. The generator runs per-compilation; it does not cross into projects that lack the Refit reference.
- The package reference is a plain `<PackageReference>` with `PrivateAssets`/`IncludeAssets` set so analyzers are excluded. Make sure the analyzer asset is included (the default `Refit` package reference includes it).
- The type is not an `interface`, or the methods have no Refit verb attribute.
- `<EnableDefaultCompileItems>false</EnableDefaultCompileItems>` or the file not being compiled in that project.

To use Refit interfaces in a separate project, add the `Refit` package reference to that project too so the generator runs there.

If you can share a minimal repro (csproj + interface) we can confirm which of these it is. Otherwise the fix is almost always "add the Refit package reference to the project that contains the interface".
